internal/datasource/os/files_test.go

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package os
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/nginx/agent/v3/api/grpc/instances"
+	"github.com/stretchr/testify/assert"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestWriteFile(t *testing.T) {
+	filePath := "/tmp/test.conf"
+	fileContent := []byte("location /test {\n    return 200 \"Test location\\n\";\n}")
+
+	err := WriteFile(fileContent, filePath)
+	assert.NoError(t, err)
+	assert.FileExists(t, filePath)
+
+	data, err := os.ReadFile(filePath)
+	assert.NoError(t, err)
+	assert.Equal(t, fileContent, data)
+
+	err = os.Remove(filePath)
+	assert.NoError(t, err)
+	assert.NoFileExists(t, filePath)
+}
+
+func TestReadCache(t *testing.T) {
+	instanceId, err := uuid.Parse("aecea348-62c1-4e3d-b848-6d6cdeb1cb9c")
+	assert.NoError(t, err)
+	cachePath := fmt.Sprintf("/tmp/%v/cache.json", instanceId.String())
```

Comment:
```suggestion
	cachePath := fmt.Sprintf("/tmp/%s/cache.json", instanceId.String())
```

Reply:
More context for cryptic percent string formatting: https://gobyexample.com/string-formatting

Reply:
I think I got them all let me know if I missed some